mlflow/pipelines/steps/predict.py

```diff
@@ -0,0 +1,153 @@
+import logging
+import os
+import time
+from typing import Dict, Any
+
+import mlflow
+from mlflow.exceptions import MlflowException, BAD_REQUEST, INVALID_PARAMETER_VALUE
+from mlflow.pipelines.cards import BaseCard
+from mlflow.pipelines.step import BaseStep
+from mlflow.pipelines.utils.execution import get_step_output_path
+from mlflow.pipelines.utils.step import get_pandas_data_profile
+from mlflow.utils._spark_utils import _get_active_spark_session
+
+_logger = logging.getLogger(__name__)
+
+
+# This should maybe imported from the preprocessing step for consistency
+_INPUT_FILE_NAME = "dataset.parquet"
+_SCORED_OUTPUT_FILE_NAME = "scored.parquet"
+
+
+class PredictStep(BaseStep):
+    def __init__(self, step_config: Dict[str, Any], pipeline_root: str):
+        super().__init__(step_config, pipeline_root)
+
+        self.run_end_time = None
+        self.execution_duration = None
+
+    def _build_profiles_and_card(self, scored_df) -> BaseCard:
+        # Build profiles for input dataset, and train / validation / test splits
+        scored_data_profile = get_pandas_data_profile(
```

Comment:
Only compute profile if not skipped: see https://github.com/mlflow/mlflow/pull/6456

Reply:
Fixing.